### ISSUE TITLE
File system permissions error on Android 14 (Oxygen OS)

### ISSUE THREAD

**Issue by clintonium-119** [BinaryQuantumSoul/esde_android_apps#3]
Hi.  I'm getting a `FileSystemException` error on my OnePlus 12R.

ES-DE is able to write to `/emulated/storage/0/ROMs`, presumably because it requests the 'All Files Access' permission, but this app returns:

`OS Error: Read-only file system, errno - 30`

Any ideas of what I can do to provide write access?  App Info shows that it doesn't request any permissions, so I don't see a way to allow it.   

NOTE: When I select the path in settings, it does prompt me for permission to write to the selected folder, but it doesn't actually grant it, it would seem.

**Comment by clintonium-119** [comment]
Just to add - I just tried the original ES Applauncher, and it did ask for the 'All Files Access' permission and it's working correctly for me.

**Comment by chocoblin** [comment]
I am having the same problem on my rg405m gamma os. both apps fail to make shortcuts in esde app, and this one mentioned it is error no. 13

**Comment by mindovermiles262** [comment]
Same issue here as with @chocoblin -- Anbernic RG405M with GammaOS. Tried with `ES Launcher` and this `ESDE Android Apps` and neither are able to create the directories/files needed.

```
Error saving apps: PathAccessException: Cannot create file, path = '/storage/<ID>/00-ROMs/androidapps/AppName.app' (OS Error: Operation not permitted, errno = 1)
```

**Comment by mindovermiles262** [comment]
As a workaround I've found you can manually create all the "Apps" you need.

Create `App Name.app` files on your computer with the filename being that of the app, and the contents of the file the AppID of the program.

For example, to get the the Android Settings as an App in ES DE, create a file named `Settings.App` and the contents of the file being

```
com.android.settings
```

Upload this file to the `ROMS/androidapps` directory and it will appear as an Android App

**Comment by DragonDreamVNY** [comment]
Similar issue on AyaNeo PocketAir 
running Android 12.
Tried on both the internal storage and on MicroSD card

When selecting paths (directory name check is disabled as my roms folder is lowercase it seems) I do get the popup about Permissions which I press Allow.. 

Path access exception errno = 1


Another thing it seems when selecting Uninstall unlisted apps, the lost goes the opposite from Show to Hide... So anything checked previously are now needing to be unchecked as it's the opposite.

**Comment by BinaryQuantumSoul** [comment]
Can anyone check if [latest pre-release](https://github.com/BinaryQuantumSoul/esde_android_apps/releases/tag/v1.3.2-beta) solve the issue? As I never experienced it on my end.

@DragonDreamVNY  @clintonium-119 @mindovermiles262 @chocoblin @LoloGX

**Comment by ghost** [comment]
I tried the latest pre-release but for me the issue is still there.
  
But I think my setup is different than the OP.
I'm using MagicX Zero28, which runs Android 10.
My ES-DE directory is in the internal storage, while the ROMs directory is in another SD card.

Also I had to manually give storage permission from app info.

![Image](https://github.com/user-attachments/assets/454b305c-34bf-48bd-8042-2dbdcc2fe3cb)

**Comment by BinaryQuantumSoul** [comment]
@Le0xFF could you try [latest](https://github.com/BinaryQuantumSoul/esde_android_apps/releases/tag/v1.3.2-beta.2)?

**Comment by ghost** [comment]
> [@Le0xFF](https://github.com/Le0xFF) could you try [latest](https://github.com/BinaryQuantumSoul/esde_android_apps/releases/tag/v1.3.2-beta.2)?

With this version everything seems to be fixed, thank you so much!
It's always nice to see open source alternatives 😄 

But I don't see "Settings" app and other system ones (Files, Gallery, Clock and similar), which were visible until version 1.3.1. In the previous beta they were not also visible.

**Comment by BinaryQuantumSoul** [comment]
> With this version everything seems to be fixed, thank you so much!
> It's always nice to see open source alternatives 😄 

You're welcome 😄
I am just very busy and slow to fix things lol

> But I don't see "Settings" app and other system ones (Files, Gallery, Clock and similar), which were visible until version 1.3.1. In the previous beta they were not also visible.

Oh man those storage permissions are hell

**Comment by ghost** [comment]
At least the problem is getting closer to be completely fixed 😄

**Comment by BinaryQuantumSoul** [comment]
> At least the problem is getting closer to be completely fixed 😄

Try [this](https://github.com/BinaryQuantumSoul/esde_android_apps/releases/tag/v1.3.3)

**Comment by ghost** [comment]
Yup, all apps are definitely there this time, issue is fixed for me!

I suggest using the whitelist option when all system apps are shown, to avoid disabling each one by one.

Thank you again!
